src/hooks/useAllTokens.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+
+import { type EdgeAccount, type EdgeCurrencyConfig, type EdgeTokenMap } from 'edge-core-js'
+
+import { useEffect, useState } from '../types/reactHooks'
+
+type EdgeTokenMaps = { [pluginId: string]: EdgeTokenMap }
+
+/**
+ * Subscribes to all the tokens across all plugins in an account.
+ */
+export function useAllTokens(account: EdgeAccount): EdgeTokenMaps {
+  // Gather the tokens from all the plugins:
+  const [out, setOut] = useState<EdgeTokenMaps>(() => {
+    const out: EdgeTokenMaps = {}
+    for (const pluginId of Object.keys(account.currencyConfig)) {
+      out[pluginId] = getAllTokens(account.currencyConfig[pluginId])
+    }
+    return out
+  })
+
+  // Watch for token changes on any of the plugins:
+  useEffect(() => {
+    const cleanups: Array<() => void> = []
+
+    for (const pluginId of Object.keys(account.currencyConfig)) {
+      const currencyConfig = account.currencyConfig[pluginId]
+      function update(): void {
+        setOut(out => ({ ...out, [pluginId]: getAllTokens(currencyConfig) }))
+      }
+      cleanups.push(currencyConfig.watch('builtinTokens', update))
+      cleanups.push(currencyConfig.watch('customTokens', update))
+    }
+
+    return () => cleanups.forEach(cleanup => cleanup())
+  }, [account])
```

Comment:
I was trying to create a useCoreWatcher which receives:
1. Watchable Core Object
2. List of keys to watch for

And returns a ref'ed object with the watched values 

But I kept running into issues with types.
Is there a chance you can+want to do it?
Watching over a Core object (especially an account and a wallet) is a super common pattern we have which is already used in a lot of places but it's going to be everywhere now that we are slowly getting rid of the GUI wallet.

Reply:
It is pretty easy with TypeScript, and already exists in the login UI. I don't believe it's possible with Flow, unfortunately, at least in a generic way.

The best we can do are individual ones like `const item = useWalletWatcher(wallet, 'item')` or `const another = useAccountWatcher(account, 'another')`.

Reply:
I'm actually pretty fine with the individual approach for now:
`const item = useWalletWatcher(wallet, 'item')`

But let's keep it for a later PR so we close this one. And I'll take a look at what you did in the login-ui